.github/workflows/build.yml

```diff
@@ -10,18 +10,20 @@ jobs:
       fail-fast: false
       matrix:
         # TODO: sanitize: [On, Off]
-        os: [ubuntu-24.04]
+        os: [ macos-15]
```

Comment:
For macOS, given the new matrix on main, we should probably just include a Debug + Release job. All of the analysis, codecov, and sanitization can happen on Ubuntu. We have 5 build jobs (1 lint) right now, so this should put us at 7 build jobs.

Reply:
I want to have a re-think on adjusting the pipeline to better streamline the steps while still handling the Debug/Release steps.

Are you sure there's no value in having macOS also do those?

Reply:
I don't think there's *no value* in macOS doing sanitization, but I'm also wary of exploding our CI jobs. If you'd like to try with all the jobs first, we can do that.